ledger-api/grpc-definitions/com/daml/ledger/api/v1/transaction.proto

```diff
@@ -65,39 +65,7 @@ message TreeEvent {
     oneof kind {
         CreatedEvent created = 1;
         ExercisedEvent exercised = 2;
+        TransferredOut transffered_out = 3;
+        TransferredIn transffered_in = 4;
```

Comment:
Transfer events currently cannot appear inside a transaction. I would therefore not consider them a `TreeEvent`. Also, some of the fields of a `TransactionTree` don't make sense for transfers (e.g. `effective_at`), so there's no point in shipping them as part of a `TransactionTree`. I'd rather introduce a new message next to `TransactionTree` for transfer events.

Reply:
Correctly observed, and agreed, this is a stretch to put in here, and we agree we need to pull the TransactionTree apart to something which has events, and to something else which has a transferout or transferin.
I would also argue to have a shared part, which we share in both Transfers and TransactionTree.
We thought it would be simpler to first introduce transfers as plain events in first iteration, due to it's simplicity, and solidify the final structure as from usage experience we build deeper understanding on what we really need for transfer related transactions -- I think we are happy to reconsider this and do this already, if we decide so this is needed already. (I have no strong opinion)